catalog/db_branch_scanner.go

```diff
@@ -1,23 +1,30 @@
 package catalog
 
 import (
+	"strconv"
+	"strings"
+
 	sq "github.com/Masterminds/squirrel"
 	"github.com/treeverse/lakefs/db"
 )
 
-const DBScannerDefaultBufferSize = 1024
+const (
+	DBScannerDefaultBufferSize = 1024
+	MaxCommitsInFilter         = 100
+)
 
 type DBBranchScanner struct {
-	opts     DBScannerOptions
-	tx       db.Tx
-	branchID int64
-	commitID CommitID
-	buf      []*DBScannerEntry
-	idx      int
-	after    string
-	ended    bool
-	err      error
-	value    *DBScannerEntry
+	opts         DBScannerOptions
+	tx           db.Tx
+	branchID     int64
+	commitID     CommitID
+	commitsWhere string
```

Comment:
we use squerrel to build the query in this scanner - we can sq.Sqlizer to hold where part expression